src/ui/public/kuery/lib/functions/is.js

```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash';
+import * as literal from '../node_types/literal';
+import { getPhraseScript } from 'ui/filter_manager/lib/phrase';
+
+export function buildNodeParams(fieldName, value, serializeStyle = 'operator') {
+  return {
+    arguments: [literal.buildNode(fieldName), literal.buildNode(value)],
+    serializeStyle
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  const { arguments:  [ fieldNameArg, valueArg ] } = node;
+  const fieldName = literal.toElasticsearchQuery(fieldNameArg);
+  const field = indexPattern.fields.byName[fieldName];
+  const value = !_.isUndefined(valueArg) ? literal.toElasticsearchQuery(valueArg) : valueArg;
+
+  if (field && field.scripted) {
+    return {
+      script: {
+        ...getPhraseScript(field, value)
+      }
+    };
+  }
+  else if (fieldName === '*' && _.isUndefined(value)) {
+    return { match_all: {} };
+  }
+  else if (fieldName === '*' && !_.isUndefined(value)) {
+    const userQuery = String(value);
+    const query = isDoubleQuoted(userQuery) ? userQuery : `"${userQuery}"`;
+
+    return {
+      simple_query_string: {
+        query,
+        all_fields: true
+      }
+    };
+  }
+  else if (fieldName !== '*' && _.isUndefined(value)) {
+    return {
+      exists: { field: fieldName }
+    };
+  }
+  else {
+    return {
+      match_phrase: {
+        [fieldName]: value
+      }
+    };
+  }
+}
+
+export function toKueryExpression(node) {
+  if (node.serializeStyle !== 'operator') {
```

Comment:
`foo is bar` isn't valid. I think of `:` as a binary operator. I think my use of the term `shorthand` for the range alias `:[]` may be confusing things here. I didn't use the term `operator` there because it's a weird syntax that sort of half wraps the arguments, I don't think most people would think of it as an operator. But we could call it an operator and get rid of the `shorthand` term entirely, if you think that would be less confusing.